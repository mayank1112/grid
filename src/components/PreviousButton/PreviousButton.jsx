import { Button } from '@material-ui/core';
import PropTypes from 'prop-types';
import { memo } from 'react';

const PreviousButton = ({ onClick, disabled }) => {
    const customProps = disabled ? { disabled: true } : {};
    return (
        <div className='button'>
            <Button {...customProps} onClick={onClick} variant="contained" color="primary">Previous</Button>
        </div>
    );
};

PreviousButton.propTypes = {
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.bool.isRequired,
};

export default memo(PreviousButton);
