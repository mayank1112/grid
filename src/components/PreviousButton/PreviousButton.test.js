import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import PreviousButton from './PreviousButton';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
const mockClick = jest.fn();

it('<PreviousButton> component should match snapshot when button is enabled', () => {
    const wrapper = shallow(<PreviousButton onClick={mockClick} disabled={false} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('<PreviousButton> component should match snapshot when button is disabled', () => {
    const wrapper = shallow(<PreviousButton onClick={mockClick} disabled={true} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('<PreviousButton> should call onClick function on click of button', () => {
    const wrapper = mount(<PreviousButton onClick={mockClick} disabled={false} />);
    const button = wrapper.find('button');
    button.simulate('click');
    expect(mockClick).toHaveBeenCalledTimes(1);
});
