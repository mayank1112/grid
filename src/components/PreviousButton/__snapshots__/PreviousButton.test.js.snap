// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviousButton> component should match snapshot when button is disabled 1`] = `
<div
  className="button"
>
  <WithStyles(ForwardRef(Button))
    color="primary"
    disabled={true}
    onClick={[MockFunction]}
    variant="contained"
  >
    Previous
  </WithStyles(ForwardRef(Button))>
</div>
`;

exports[`<PreviousButton> component should match snapshot when button is enabled 1`] = `
<div
  className="button"
>
  <WithStyles(ForwardRef(Button))
    color="primary"
    onClick={[MockFunction]}
    variant="contained"
  >
    Previous
  </WithStyles(ForwardRef(Button))>
</div>
`;
